### PR TITLE
Add Learnings from Clean Coder article

### PR DESCRIPTION
**Pull request by 2KAbhishek**

### CODE REVIEW

File: content/blog/learnings-from-clean-coder.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+slug = "demo-post"
+date = 2022-04-29T11:55:43+05:30
+image = "/images/2021/07/demo.jpg"
+draft = true
+author = "Incubyte"
+authorLink = "https://incubyte.co"
+description = ""
+tags = ["Playbook", "Software Craftsmanship"]
+categories = ["Playbook", "Software Craftsmanship"]
+type = ""
```
Comment (utsav00) [inline content/blog/learnings-from-clean-coder.md:12]: Why is this empty?

Comment (utsav00) [reply content/blog/learnings-from-clean-coder.md:12]: For demo changes

File: content/blog/learnings-from-clean-coder.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
++++
+
+## Learnings from Robert “Uncle Bob” Martin
+
+Robert Martin, fondly known as “Uncle Bob” in the software community, is a software engineer and more famously an author and the founding fathers of many software movements like software craftsmanship, agile methodology and test-driven development. Some of his most famous works include Clean Code, The Clean Coder, Clean Architecture, and Clean Agile. Martin is also a frequent instructor, with the ability to boil down difficult topics into simple, understandable format, along with an occasional cheeky humour.
```
Comment (rushaliparikh) [inline content/blog/learnings-from-clean-coder.md:17]: Make sure you're checking your writing using Grammarly before you create a PR

File: content/blog/learnings-from-clean-coder.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+### 2. Duplication is the Evil
+
+Uncle Bob goes on a length about various characteristics of bad code. Amongst them, code duplication holds a special spot, because majority of code problems have their origins linked to duplication. It has inspired many coding patterns, techniques and ideologies. For e.g., the database normalization techniques (1NF, 2NF, 3NF…) by Codd is ultimately about removing redundancy in the database design. In fact, paradigms like OOP, AOP, COP, Structured Programming, are all different ways to deal with duplication. So, one of the primary tasks while writing a piece of software is to be mindful about code duplication.
```
Comment (rushaliparikh) [inline content/blog/learnings-from-clean-coder.md:27]: Isn't it also true that some level of duplication may be fine depending on the reason and context? If yes, maybe you want to add it here?

File: content/blog/learnings-from-clean-coder.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+Writing tests is often seen as something which takes a lot of time and doesn’t serve a real purpose.
+
+>*“I wrote the code; I know it works… Don’t you trust me?”*
+
+However, that’s far from the truth. Writing tests unravels many of the aspects we would have otherwise not uncovered. A good test suite could be a challenge to write, but the F.I.R.S.T. principle can help us out:
+
+* Fast – Tests should run fast. If they are slow, nobody would want to run them.
+* Independent – Each test should be independent of other test, and they should be isolated enough to be run individually.
+* Repeatable – They should be environment independent. Should run on dev, prod, QA, etc.
+* Self-Validating – They test should either pass or fail. Anything in between means more time would be wasted in reading logs and stuff.
```
Comment (rushaliparikh) [inline content/blog/learnings-from-clean-coder.md:54]: Can you find a way to say the same thing without using the word 'stuff'? Another general rule, avoid ambiguity.

**Comment by 2KAbhishek** [comment]
Thanks for the review, I haven't edited it, just the first version.
Your insights will make it much easier! 😆

**Comment by akshayvadher** [comment]
Nice image!

**Review by ArohiParikh — APPROVED** [review]

**Pull request merged**